geth/node/manager.go

```diff
@@ -518,3 +519,77 @@ func (m *NodeManager) isNodeAvailable() error {
 
 	return nil
 }
+
+// SyncAndStopNode waits until node synchronization and exits.
+// Reports errors to returned channel.
+func (m *NodeManager) SyncAndStopNode(timeout time.Duration) (errChan <-chan error) {
```

Comment:
I don't see any benefits in having a single method that does all this stuff. There is a method to stop a node so I guess only `Sync()` would be needed. In `main.go`, it can be called like this:
```go
if err := m.Sync(); err != nil {
  // exit
}
if err := m.StopNode(); err != nil {
  // exit
}
```

Reply:
Also, it should have a synchronous interface. If the caller wants to call it asynchronously, then it's its trouble.